app/controllers/api/v1/offers_controller.rb

```diff
@@ -163,11 +163,12 @@ def filter_created_by(offers)
       end
 
       def offer_summary_response(records)
+        records = params["shallow"]== 'true' ? records : records.with_summary_eager_load
+        serializer = params[:shallow] == 'true' ? shallow_summary_serializer : summary_serializer
```

Comment:
Could you switch the logic around and replace `params['shallow']` with params['companies']. The default should be the summary_serializer and the offer_company_serializer can be used if params['companies'] == 'true'
Replace `shallow_summary_serializer` with `offer_company_serializer` to be more explicit about what we are returning. Makes it much clearer.